### PR TITLE
Original: 开源，社区与朋友们-2019 KubeCon + ClondNativeCon + Open Source Summit有感

### PR DESCRIPTION
**Pull request by zhaohuabing**
请审校者在合并该 PR 前需要检查以下内容：

- [x] Header 设置正确，包括文章的类型、tag、分类、作者、译者、摘要、reviewer 发布时间等信息
- [ ] 关联 Issue 编号 [#${ISSUE_ID}](https://github.com/servicemesher/website/issues/${ISSUE_ID})
- [x] 该 PR Assigned 给了提交者
- [x] 指定了 Reviewer
- [x] 为 PR 明确了 label 信息，如 categories、kind、size、tag、status
- [x] 所有代码片段都指定了代码语言
- [ ] 翻译的文章添加了**编者按**
- [x] 文章遵守了路径规范  `YEAR/MONTH/TITLE/index.md`，例如 `2019/04/new-blog/index.md`
- [x] 图片保存了在 GitHub上，跟文章在同级目录
- [x] 文章标题简短且语义明确
- [x] 文章语句通顺，没有语病和格式问题
- [x] 文中没有不可达的链接
- [ ] 提交者同意授权 ServiceMesher 公众号和 [http://www.servicemesher.com](http://www.servicemesher.com) 发布我原创或翻译的文章

当所有以上问题都确定之后，可以由管理者合并 PR 并关闭关联 issue。

### CODE REVIEW

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+大会第二天的Keynote一直是Linux基金会宣传其理念，愿景以及赞助商进行市场宣传的重要活动。但当天最让人期待的莫过于Linus在keynote后的一次嘉宾谈话。为了一睹Linus的真容，很多人一早就在举行活动的red hall前排起了长队。我由于来得较早，非常幸运地抢到了一个较好的位置，位于第六排中间，正好正对台上的演讲嘉宾。
+
+首先是Linux基金执行董事Jim Zemlin的开场演讲，他的演讲中提到几个重要的事实和数据：
+
+* Linux已经成为世界上最重要的软件平台：100%的超级计算机，82%的手机，68%的企业服务器，90%的公有云负载，62%的嵌入式操作系统都运行在Linux上
+* 中国已经成为第二大Kubernets贡献者
+* 开源代码通常占现代操作系统代码总量的80%-90%
+* 10%的CNCF会员来自中国，在会议中，宣布了蚂蚁金服充值成为CNCF黄金会员
+
+从他的演讲中可以感受到来自中国的公司和开发者在开源软件，特别是云原生领域中发挥了越来越重要的作用。由于中国的市场体量很大，中国公司在在使用开源软件时会面临一些国外难以遇到的问题，例如如何应对数以亿记的用户量。在解决这些问题的同时，很多中国公司也将这些优化和改进贡献到了上游项目，造福整个开源社区，形成了良性的互动和循环。从后面两天的主题演讲，我也看到中国公司，特别是阿里系在回馈上游开源社区方面是做的非常好的。
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: 亿记->亿计

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+title: "开源，社区与朋友们-2019 KubeCon + ClondNativeCon + Open Source Summit有感"
+date: 2019-06-28T11:05:00+08:00
+draft: false
+banner: "/img/blog/banners/kubecon-cncf-oss-2019-banner.jpg"
+author: "赵化冰"
+authorlink: "<https://zhaohuabing.com>"
+reviewer: []
+reviewerlink: []
+originallink: ""
+summary: "奇妙的 2019 KubeCon + ClondNativeCon + Open Source Summit 大会！在这里，我近距离接触了大神 Linus；见到了自 ServiceMesher 社区的很多朋友；还遇到了搞 Kubernets 的恩格斯后人！"
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+[使用CloudMesh 轻松体验ServiceMesh](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/cloudmesh.pdf)
+
+## Keynote
+
+大会第二天的Keynote一直是Linux基金会宣传其理念，愿景以及赞助商进行市场宣传的重要活动。但当天最让人期待的莫过于Linus在keynote后的一次嘉宾谈话。为了一睹Linus的真容，很多人一早就在举行活动的red hall前排起了长队。我由于来得较早，非常幸运地抢到了一个较好的位置，位于第六排中间，正好正对台上的演讲嘉宾。
+
+首先是Linux基金执行董事Jim Zemlin的开场演讲，他的演讲中提到几个重要的事实和数据：
+
+* Linux已经成为世界上最重要的软件平台：100%的超级计算机，82%的手机，68%的企业服务器，90%的公有云负载，62%的嵌入式操作系统都运行在Linux上
+* 中国已经成为第二大Kubernets贡献者
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+### Envoy的可观察性
+
+该Session的内容本身没有太多特别之处，主要是根据HashiCorp内部使用Envoy的经验讲了Envoy提供了哪些可以用于可观察性的配置/接口。
+
+我在会后和主讲人Nic Jackson聊了一下HashiCorp的Consul Connect项目，Consul Connect是一个支持Hybrid Cloud的Service Mesh项目，采用了Consul作为Service Registry，Envoy作为Sidecar，自己写的控制面，后面会兼容SMI(Service Mesh Interface）。
+
+Consul connect在进行sidecar导流时没有使用IPTables，而是将所有upstream的Service全部映射到了locahost的端口上，并通过Envoy监听Localhost来实现Sidecar流量拦截。这种方式去掉了IPtables，Nic说他们目前的测试性能非常理想。
+
+HashiCorp的方案主要是面向多云环境以及同时存在Kubernets和VM，bare metal场景的企业用户。其方案和我们目前自己使用的方案比较类似，后面值得长期关注。如果希望了解更多Consul Connect的内容，可以关注HashiCorp的博客及网站： <https://www.hashicorp.com/blog/layer-7-observability-with-consul-service-mesh>
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+![](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/get-trust-bundle.jpg)
+
+用得到的证书信息对来自另一个信任域的请求进行身份认证
+![](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/service-communication.jpg)
+
+[完整PPT链接](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/How-SPIFFE-helps-Istio-in-Service-Mesh-Federation.pdf)
+
+### IngressRoute：如何为 Kubernetes 提供更好的 Ingress 对象
+
+介绍了Kubernets Ingress的局限性和Contour的解决方案：采用一个新的CRD。Ingress的局限性一直为社区所诟病，Kubernets中的Ingress只是七层代理的一个最小集合，为了解决该问题，几乎每个Ingress Controller项目都使用了自定义的扩展，包括Annotation和自定义CRD两种方式。
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+Session后和David Cheney交流了两个问题：
+
+* 是否会把IngressRoute CRD中的特性贡献到Kubernets社区，他答复说可能会在Kubernets Ingress的下一个版本中加入部分新特性。
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+* Contour是否会和Istio进行集成，以支持Service Mesh的一些特性，例如路由控制等。他答复是很可能会和SMI集成，而不是直接和Istio绑定。
+
+其实各个具体的产品对Ingress的要求各不相同，如何在Kubernets自身API的简洁性和产品要求的功能实用性之间进行平衡是一个难题，因此该问题应该还会存在很长时间。我的建议是采用一个API Gateway + Sidecar 来代替 Kubernets Ingress，参见我之前的文章[如何为服务网格选择入口网关？](https://zhaohuabing.com/post/2019-03-29-how-to-choose-ingress-for-service-mesh/)
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+![](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/job-board.jpg)
+
+### ServiceMesher社区的朋友
+
+这次大会上见到了很多在ServiceMesher社区中聊过多次，但未曾谋面的朋友，包括小剑，Jimmy，崔总，超盟,fox ...等等, 还有好些以前不认识的同学主动过来给我打招呼说看过我的博客文章。非常高兴能有机会见到这些志同道合的朋友，也希望我们的ServiceMesher社区能越办越好，冲出中国，走向全球!（已经遇到读过我文章的外国朋友向我打听ServiceMesher社区了，他的原话是：Would you agree to say that this is the only place where such in depth material is available?)
+
+ServiceMesher社区的朋友们
+![](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/background.jpg)
+
+### 恩格斯也来搞Kubernets了！
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+最后一天中午时，大会的各个厂商开始陆陆续续拆展台了。我领了一个盒饭坐在展馆门口的台阶上吃，旁边一个国外的参会者在等他的朋友，我就和他聊了两句。他来自德国汉堡，是一个服装公司的IT工程师。目前他们公司正在向云上迁移，使用了Private cloud/public cloud的hybrid方案。前端的Web服务器放在AWS上，后端的业务应用放在公司的数据中心，采用VPN连接数据中心和AWS。从他们公司的案例可以看出，混合云方案将在非常长的时间内存在，也难怪在展会上看到了多家做多云管理服务的公司。
+
+聊了一会儿后，他指着自己胸牌上的名字问到，你知道恩格斯吗？我一时还没有回过神来，他又问了一次，你知道恩格斯吗？他在中国很有名的。我这才想起，问到：是那个搞共产主义的恩格斯吗？他兴奋地说，是的，他是我曾曾曾祖父的侄子。What a small world！没想到恩格斯家族也来搞Kubernets了。不过Kubernets的理念和恩格斯、马克思提出的那个世界大同的理想是不是也很相似呢？
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+
+![](https://raw.githubusercontent.com/servicemesher/website/master/content/blog/kubecon-cncf-oss-2019/germany-friend.jpg)
+
+## 参会的几点感想
+
+* Kubernets已经成为云上的标准，Cloud Native正当时：这个观点是阿里在演讲中提出来的，我非常赞同。从这次大会的参展厂商和技术分享中可以看到，几乎所有的云厂商/运营商/企业都已经采用了容器和Kubernets，并且在积极推进Service Mesh等云原生技术栈的实践和运用。
```
Comment (haiker2011) [inline content/blog/kubecon-cncf-oss-2019/index.md]: Kubernets->Kubernetes

**Review by haiker2011 — CHANGES_REQUESTED** [review]

**Comment by zhaohuabing** [comment]
content/blog/kubecon-cncf-oss-2019/index.md:17: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:26: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:44: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:75: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:79: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:92: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:96: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:105: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:117: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:130: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:132: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:139: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:146: MD022 Headers should be surrounded by blank lines
content/blog/kubecon-cncf-oss-2019/index.md:154: MD022 Headers should be surrounded by blank lines
规则是不是有点问题？ 标题前后是有空白行的。

**Comment by haiker2011** [comment]
@zhaohuabing 嗯嗯，我也看了一下，感觉确实有点问题，格式先忽略吧

File: content/blog/kubecon-cncf-oss-2019/index.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+title: "开源，社区与朋友们-2019 KubeCon + ClondNativeCon + Open Source Summit有感"
+date: 2019-06-28T11:05:00+08:00
+draft: false
+banner: "/img/blog/banners/kubecon-cncf-oss-2019-banner.jpg"
+author: "赵化冰"
+authorlink: "<https://zhaohuabing.com>"
+reviewer: [孙海洲]
+reviewerlink: ["<https://github.com/haiker2011>"]
+originallink: ""
+summary: "奇妙的 2019 KubeCon + ClondNativeCon + Open Source Summit 大会！在这里，我近距离接触了大神 Linus；见到了自 ServiceMesher 社区的很多朋友；还遇到了搞 Kubernetes 的恩格斯后人！"
```
Comment (SataQiu) [inline content/blog/kubecon-cncf-oss-2019/index.md]:
```suggestion
summary: "奇妙的 2019 KubeCon + ClondNativeCon + Open Source Summit 大会！在这里，我近距离接触了大神 Linus；见到了来自 ServiceMesher 社区的很多朋友；还遇到了搞 Kubernetes 的恩格斯后人！"
```

**Review by SataQiu — COMMENTED** [review]
好文！

**Comment by zhaohuabing** [comment]
还有需要修改的吗？

**Review by SataQiu — APPROVED** [review]

**Review by haiker2011 — APPROVED** [review]

**Pull request merged**